superset/models/dashboard.py

```diff
@@ -96,6 +96,7 @@ def copy_dashboard(_mapper: Mapper, connection: Connection, target: Dashboard) -
     )
     session.add(extra_attributes)
     session.commit()
+    session.close()
```

Comment:
I wonder if we can just reuse the db.session everywhere so we don't create this arbitrary session

Reply:
if we really need to use this session, better to wrap into try/catch and close at final otherwise exceptions thrown in commit will skip session.close

Reply:
@zephyring that is the ideal solution, using the same session (which is obtainable via `inspect(target).session`) however per https://github.com/apache/superset/pull/25031#discussion_r1300652358 this is currently problematic given the way our SQLAlchemy event listener callbacks are configured.

Reply:
@john-bodley Could you apply the `try/finally` pattern to the places where you're closing the session? I also think we can remove the `session.commit()` on line 91.